src/cls/queue/cls_queue_src.cc

```diff
@@ -0,0 +1,478 @@
+
+    //Process the chunk of data read
+    unsigned index = 0;
+    auto it = bl_chunk.cbegin();
+    uint64_t size_to_process = bl_chunk.length();
+    do {
+      CLS_LOG(10, "INFO: queue_list_entries(): index: %u, size_to_process: %lu\n", index, size_to_process);
+      cls_queue_entry entry;
+      it.seek(index);
+      //Populate offset if not done in previous iteration
+      if (! offset_populated) {
+        cls_queue_marker marker = {start_offset + index, gen};
+        CLS_LOG(5, "INFO: queue_list_entries(): offset: %s\n", marker.to_str().c_str());
+        entry.marker = marker.to_str();
+      }
+      // Magic number + Data size - process if not done in previous iteration
+      if (! entry_start_processed ) {
+        if (size_to_process >= (sizeof(uint16_t) + sizeof(uint64_t))) {
+          // Decode magic number at start
+          try {
+            decode(entry_start, it);
+          } catch (buffer::error& err) {
+            CLS_LOG(10, "ERROR: queue_list_entries: failed to decode entry start \n");
+            return -EINVAL;
+          }
+          if (entry_start != QUEUE_ENTRY_START) {
+            CLS_LOG(5, "ERROR: queue_list_entries: invalid entry start %u\n", entry_start);
+            return -EINVAL;
+          }
+          index += sizeof(uint16_t);
+          size_to_process -= sizeof(uint16_t);
+          // Decode data size
+          try {
+            decode(data_size, it);
+          } catch (buffer::error& err) {
+            CLS_LOG(10, "ERROR: queue_list_entries: failed to decode data size \n");
+            return -EINVAL;
+          }
+        } else {
+          // Copy unprocessed data to bl
+          bl_chunk.copy(index, size_to_process, bl);
+          offset_populated = true;
+          CLS_LOG(10, "INFO: queue_list_entries: not enough data to read entry start and data size, breaking out!\n");
+          break;
+        }
+        CLS_LOG(20, "INFO: queue_list_entries(): data size: %lu\n", data_size);
+        index += sizeof(uint64_t);
+        size_to_process -= sizeof(uint64_t);
+      }
+      // Data
+      if (data_size <= size_to_process) {
+        bl_chunk.copy(index, data_size, entry.data);
+        index += entry.data.length();
+        size_to_process -= entry.data.length();
+      } else {
+        bl_chunk.copy(index, size_to_process, bl);
+        offset_populated = true;
+        entry_start_processed = true;
+        CLS_LOG(10, "INFO: queue_list_entries(): not enough data to read data, breaking out!\n");
+        break;
+      }
+      op_ret.entries.emplace_back(entry);
+      // Resetting some values
+      offset_populated = false;
+      entry_start_processed = false;
+      data_size = 0;
+      entry_start = 0;
+      num_ops++;
+      if (num_ops == op.max) {
+        CLS_LOG(10, "INFO: queue_list_entries(): num_ops is same as op.max, hence breaking out from inner loop!\n");
+        break;
+      }
+      if (index == bl_chunk.length()) {
+        break;
+      }
```

Comment:
i don't think we need this check, since the `do {} while(index < bl_chunk.length());` will stop here anyway

Reply:
yeah ok